newton/sim/builder.py

```diff
@@ -3182,6 +3182,16 @@ def _update_body_mass(self, i, m, I, p, q):
         else:
             self.body_inv_inertia[i] = new_inertia
 
+    def add_free_joints_to_floating_bodies(self):
```

Comment:
Can you add an optional parameter for the body indices, a list of indices? It can be None by default:
```py
body_indices: list[int] | None = None
```
If None you apply this to all bodies. That way the USD importer can add these free joints only to the bodies it inserted, not every body in the ModelBuilder.

Reply:
Thanks for catching this. I implemented your idea, with `new_bodies` being an `Iterable[int]`.